applications/FSIApplication/python_scripts/fsi_coupling_interface.py

```diff
@@ -154,9 +340,30 @@ def _get_partitioned_fsi_utilities(self):
 
     def _create_partitioned_fsi_utilities(self):
         domain_size = self.GetFatherModelPart().ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
-        if domain_size == 2:
-            return KratosFSI.PartitionedFSIUtilitiesArray2D()
-        elif domain_size == 3:
-            return KratosFSI.PartitionedFSIUtilitiesArray3D()
+        if not self.GetFatherModelPart().IsDistributed():
+            if domain_size == 2:
+                if self.scalar_output:
+                    return KratosFSI.PartitionedFSIUtilitiesDouble2D()
+                else:
+                    return KratosFSI.PartitionedFSIUtilitiesArray2D()
+            elif domain_size == 3:
+                if self.scalar_output:
+                    return KratosFSI.PartitionedFSIUtilitiesDouble3D()
+                else:
+                    return KratosFSI.PartitionedFSIUtilitiesArray3D()
+            else:
+                raise Exception("Domain size expected to be 2 or 3. Got " + str(self.domain_size))
         else:
-            raise Exception("Domain size expected to be 2 or 3. Got " + str(self.domain_size))
+            self._epetra_communicator = KratosTrilinos.CreateCommunicator()
```

Comment:
... but here

Reply:
Pf... As we have experienced in the past some issues with the MPI imports, I'd like to keep it in the top of the file.

Reply:
I think those issues should be solved now but up to you

but be aware that as we discussed the GetDefault might be moved to the testing namespace so maybe it would be better to check the Kernel if it is a distributed run